test_rmw_implementation/test/test_serialize_deserialize.cpp

```diff
@@ -141,6 +214,49 @@ TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), clean_round_trip
     rmw_get_error_string().str;
 }
 
+TEST_F(
+  CLASSNAME(
+    TestSerializeDeserialize,
+    RMW_IMPLEMENTATION), clean_round_trip_for_cpp_bounded_message) {
+  using test_message = test_msgs::msg::BoundedPlainSequences;
+  const rosidl_message_type_support_t * ts =
+    rosidl_typesupport_cpp::get_message_type_support_handle<test_message>();
+  test_message input_message{};
+  test_message output_message{};
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+
```

Comment:
nit, but i would also use `OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT`.

```suggestion
  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
  {
    EXPECT_EQ(
      RMW_RET_OK, rmw_serialized_message_fini(
        &serialized_message)) << rmw_get_error_string().str;
  });
```

Reply:
done

Reply:
@MiguelCompany nit: check block indentation